plugins/destination/mysql/client/client.go

```diff
@@ -0,0 +1,65 @@
+package client
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+
+	mysql "github.com/go-sql-driver/mysql"
+)
+
+type Client struct {
+	destination.UnimplementedUnmanagedWriter
+	destination.DefaultReverseTransformer
+	logger zerolog.Logger
+
+	spec      specs.Destination
+	mySQLSpec Spec
+
+	db *sql.DB
+}
+
+func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (destination.Client, error) {
+	var mySQLSpec Spec
+	err := spec.UnmarshalSpec(&mySQLSpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal spec: %w", err)
+	}
+
+	mySQLSpec.SetDefaults()
+	if err := mySQLSpec.Validate(); err != nil {
+		return nil, err
+	}
+
+	dsn, err := mysql.ParseDSN(mySQLSpec.ConnectionString)
+	if err != nil {
+		return nil, fmt.Errorf("invalid MySQL connection string: %w", err)
+	}
+	if dsn.Params == nil {
+		dsn.Params = map[string]string{}
+	}
+	dsn.Params["parseTime"] = "true"
+	db, err := sql.Open("mysql", dsn.FormatDSN())
+	if err != nil {
+		return nil, fmt.Errorf("failed to open mysql connection: %w", err)
+	}
+
+	db.SetConnMaxLifetime(time.Minute * 3)
+	db.SetMaxOpenConns(10)
```

Comment:
maybe we can move this to env/spec?

Reply:
I would only add additional configuration based on users requests